### PR TITLE
fix(sdk): stop evaluated text from splicing the text-complexity report template

### PR DESCRIPTION
**Pull request by adnanrhussain**
Found while reviewing comments, but this is a code defect, not a comment.

`formatAsHTML` spliced its payload into the report template with a **string** replacement:

```ts
return reportTemplate.replace(INJECTION_MARKER, `var REPORT_DATA = ${safeJson};`);
```

In a replacement string, `$'` means "everything after the match" — here the whole template tail, including its literal `</script>`. Evaluated text and model reasoning both reach this payload, so a passage containing `$'` corrupts the report:

```
clean   </script> count: 1
hostile </script> count: 3     ← the template tail spliced in, twice
```

Measured against the built package with `text` = `A passage containing $' a dollar-quote.` — the same text also appears in `reasoning`, hence twice. After the fix: **1 and 1**.

The escaping above it was real but incomplete: it neutralised `<`, `>` and `&`, and its comment claimed that "prevents `</script>` injection" — which the string replacement then reintroduced downstream.

## The fix

Both report formatters now share one guarded injection. `standards-output.ts` already had the right implementation and a comment explaining exactly this hazard; `formatters.ts` had neither. Rather than copy it, `injectReportData` and `toInlineJson` move to `batch/report-injection.ts` and both call it — so there is one implementation to be right.

That also gives the text-complexity report the U+2028/U+2029 escaping it was missing: valid JSON, but line terminators to a pre-ES2019 parser.

## Why it went unnoticed

Two reviewers flagged it independently. My first attempt to verify it **produced a false negative** — I measured the length of the `var REPORT_DATA = …` line, but the spliced tail contains newlines, so the payload line got *shorter*. Counting `</script>` is what shows it.

The regression test counts closing tags and, separately, parses the payload back. The parse matters: `` $` `` and `$$` corrupt the JSON without adding a closing tag, so a tag-count-only test would pass while the report rendered empty. With the payload check, **5 of the 10 tests fail against the pre-fix code**; with tag counting alone it was 3.

## Validation

- `typecheck`, `lint`, `test:unit` (1064), `build`, `scripts/check.py`
- Reverted the fix and confirmed 5 tests fail; restored and confirmed all pass
- Re-ran the original exploit against `dist/` and confirmed the closing-tag count no longer changes
- Standards' existing injection tests pass unchanged, re-pointed at the shared module

One thing I did **not** change: `report-template.html` also has `REPORT_DATA = REPORT_DATA || MOCK_REPORT_DATA`, so a null payload would render 300 fabricated rows as if real. I checked — injection always produces a real object, so that path is defensive rather than live. The comment above it claiming the mock "is never referenced when real data has been injected" is wrong and is fixed in the comments PR that follows.

### CODE REVIEW

File: sdks/typescript/src/batch/report-injection.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * so they cannot break the script.
+ */
+export function toInlineJson(data: unknown): string {
+  return JSON.stringify(data)
+    .replace(/</g, '\\u003c')
+    .replace(/>/g, '\\u003e')
+    .replace(/&/g, '\\u0026')
+    .replace(/\u2028/g, '\\u2028')
+    .replace(/\u2029/g, '\\u2029');
+}
```
[the remark this answers is not in the document]
Comment (adnanrhussain) [reply sdks/typescript/src/batch/report-injection.ts]: Correct on the mechanism, fixed at the type level.

```
JSON.stringify(undefined) = undefined
JSON.stringify(()=>{})    = undefined
then .replace -> TypeError: Cannot read properties of undefined (reading 'replace')
```

Not reachable today — `toInlineJson` isn't exported from the package index, and both callers pass a constructed report object — so this is a latent hazard rather than a live bug.

Narrowed the parameter to `Record<string, unknown>` rather than adding a runtime guard, so misuse fails to compile:

```
error TS2345: Argument of type 'undefined' is not assignable to parameter of type 'Record<string, unknown>'
```

A runtime check would be defensive code on a path no caller can reach; the compiler covers it for free. One honest limit: a function is still structurally assignable to `Record<string, unknown>`, so that specific misuse would compile — the realistic case (`undefined`, a primitive, a missing field) is blocked.

File: sdks/typescript/tests/unit/batch/report-injection.test.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  });
+
+  it('keeps the payload parseable when text carries a line separator', () => {
+    // U+2028 is valid JSON but a line terminator to a pre-ES2019 parser.
+    const html = reportFor('A passage with 
 inside.');
+
+    expect(html).toContain('\\u2028');
+    expect(html).not.toContain('
');
```
[the remark this answers is not in the document]
Comment (adnanrhussain) [reply sdks/typescript/tests/unit/batch/report-injection.test.ts]: Fixed — and you caught a silent failure, not just a style issue.

I had already tried to replace those literals with escapes earlier in this PR, but I used `replace` without asserting the match, so it no-op'd and I didn't notice. Both lines still held raw U+2028:

```
line 92 | literal-separator=True | "const html = reportFor('A passage with   inside.');"
line 95 | literal-separator=True | "expect(html).not.toContain(' ');"
```

Now explicit escapes, with an assertion that no literal survives:

```
line 92 | literal-separator=False | "const html = reportFor('A passage with \\u2028 inside.');"
line 95 | literal-separator=False | "expect(html).not.toContain('\\u2028');"
```

Re-confirmed the test is still load-bearing rather than cosmetically correct: disabling the U+2028 escaping in `toInlineJson` still fails it.
